packages/eslint-plugin/src/rules/strict-comparisons.ts

```diff
@@ -0,0 +1,314 @@
+      );
+
+      if (overlappingTypes.length > 0) {
+        return getStrictestKind(overlappingTypes);
+      } else {
+        // In case one of the types is "any", get the strictest type of the other array
+        if (arrayContainsKind(typesLeft, [TypeKind.Any])) {
+          return getStrictestKind(typesRight);
+        }
+        if (arrayContainsKind(typesRight, [TypeKind.Any])) {
+          return getStrictestKind(typesLeft);
+        }
+
+        // In case one array contains Null or Undefined and the other an Object, return Object
+        if (
+          (arrayContainsKind(typesLeft, [TypeKind.Null, TypeKind.Undefined]) &&
+            arrayContainsKind(typesRight, [TypeKind.Object])) ||
+          (arrayContainsKind(typesRight, [TypeKind.Null, TypeKind.Undefined]) &&
+            arrayContainsKind(typesLeft, [TypeKind.Object]))
+        ) {
+          return TypeKind.Object;
+        }
+        return undefined;
+      }
+    }
+
+    /**
+     * Check if a specific TypeKind is present in an array
+     * @param types array of TypeKinds
+     * @param typeToCheck TypeKind to check
+     */
+    function arrayContainsKind(
+      types: TypeKind[],
+      typeToCheck: TypeKind[],
+    ): boolean {
+      return types.some(type => typeToCheck.includes(type));
+    }
+
+    /**
+     * Return the strictest kind of an array
+     * @param types array of TypeKinds
+     */
+    function getStrictestKind(types: TypeKind[]): TypeKind {
+      return Math.max.apply(Math, types);
+    }
+
+    /**
+     * Check if the operator is a comparison operator
+     * @param operator the operator to check
+     */
+    function isComparisonOperator(operator: string): boolean {
+      if (isEqualityOperator(operator)) {
+        return true;
+      }
+      switch (operator) {
+        case '<':
+        case '>':
+        case '<=':
+        case '>=':
+          return true;
+        default:
+          return false;
+      }
+    }
+
+    /**
+     * Check if the operator is an equality operator
+     * @param operator the operator to check
+     */
+    function isEqualityOperator(operator: string): boolean {
+      switch (operator) {
+        case '==':
+        case '!=':
+        case '===':
+        case '!==':
+          return true;
+        default:
+          return false;
+      }
+    }
+
+    /**
+     * Helper function to get base type of node
+     * @param node the node to be evaluated.
+     */
+    function getNodeType(node: TSESTree.Node): ts.Type {
+      const tsNode = service.esTreeNodeToTSNodeMap.get(node);
+      return typeChecker.getTypeAtLocation(tsNode);
+    }
+
+    return {
+      BinaryExpression(node: TSESTree.BinaryExpression) {
+        if (isComparisonOperator(node.operator)) {
+          const leftType = getNodeType(node.left);
+          const rightType = getNodeType(node.right);
+
+          const leftKinds: TypeKind[] = getKinds(leftType);
+          const rightKinds: TypeKind[] = getKinds(rightType);
```

Comment:
you don't need the explicit typecast here because `getKinds` has an explicit function return type.